sdk/eventgrid/eventgrid/README.md

```diff
@@ -1,119 +1,210 @@
+
+```js
+const { EventGridClient, EventGridSharedAccessSignatureCredential } = require("@azure/eventgrid");
+
+const client = new EventGridClient(
+  "<endpoint>",
+  new EventGridSharedAccessSignatureCredential("<SAS Token>")
+);
+```
+
+You can generate a SAS token by using the `generateSharedAccessSigniture` instance method on the `EventGridClient` type. Because the Topic's Access Key is used as part of generating the SAS token, you need to create the `EventGridClient` using an Access Key:
+
+```js
+const { EventGridClient, AzureKeyCredential } = require("@azure/eventgrid");
+
+const client = new EventGridClient("<endpoint>", new AzureKeyCredential("<API key>"));
+
+// Create a SAS Token which expires on 2020-01-01 at Midnight.
+const token = client.generateSharedAccessSignature(
+  generateSharedAccessSignature(new Date(2020, 0, 1, 0, 0, 0))
+);
 ```
 
-##### Sample code
-
-See https://github.com/Azure/ms-rest-browserauth to learn how to authenticate to Azure in the browser.
-
-- index.html
-```html
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-    <title>@azure/eventgrid sample</title>
-    <script src="node_modules/@azure/ms-rest-js/dist/msRest.browser.js"></script>
-    <script src="node_modules/@azure/ms-rest-azure-js/dist/msRestAzure.js"></script>
-    <script src="node_modules/@azure/ms-rest-browserauth/dist/msAuth.js"></script>
-    <script src="node_modules/@azure/eventgrid/dist/eventgrid.js"></script>
-    <script type="text/javascript">
-      const subscriptionId = "<Subscription_Id>";
-      const authManager = new msAuth.AuthManager({
-        clientId: "<client id for your Azure AD app>",
-        tenant: "<optional tenant for your organization>"
-      });
-      authManager.finalizeLogin().then((res) => {
-        if (!res.isLoggedIn) {
-          // may cause redirects
-          authManager.login();
-        }
-        const client = new Azure.Eventgrid.EventGridClient(res.creds, subscriptionId);
-        const topicHostname = "testtopicHostname";
-        const events = [{
-          id: "testid",
-          topic: "testtopic",
-          subject: "testsubject",
-          data: {},
-          eventType: "testeventType",
-          eventTime: new Date().toISOString(),
-          dataVersion: "testdataVersion"
-        }];
-        client.publishEvents(topicHostname, events).then((result) => {
-          console.log("The result is:");
-          console.log(result);
-        }).catch((err) => {
-          console.log("An error occurred:");
-          console.error(err);
-        });
-      });
-    </script>
-  </head>
-  <body></body>
-</html>
+## Key concepts
+
+### EventGridClient
+
+`EventGridClient` is used sending events to an Event Grid Topic or an Event Grid Domain.
+
+### Event Schemas
```

Comment:
Is it worth noting that this is a top-level schema, and that users can apply custom schemas (with custom serialization!) to EG event payloads?